File: 2021/elixir/apps/test_data/lib/test_data.ex
defmodule TestData do
  @test_data_dir Path.join(File.cwd!(), "../../../test_data")

  def get(day, part) do
    Path.join([@test_data_dir, day, part <> ".yaml"])
    |> YamlElixir.read_all_from_file()
    |> case do
      {:ok, data} -> data
      {:error, error} -> raise ~s("Read test data "#{day}/#{part}.yaml" failed: #{error.message}")
    end
  end
end

File: 2021/go/testdata/testdata.go
package testdata

import (
	"bytes"
	"errors"
	"fmt"
	"io"
	"io/ioutil"
	"path/filepath"
	"runtime"

	"gopkg.in/yaml.v3"
)

type TestCase struct {
	Input  string `yaml:"input"`
	Output string `yaml:"output"`
}

func Get(day int, part int) (error, []TestCase) {
	file := filepath.Join(cwd(), fmt.Sprintf("../../test_data/day%d/part%d.yaml", day, part))
	content, err := ioutil.ReadFile(file)
	if err != nil {
		return err, nil
	}

	decoder := yaml.NewDecoder(bytes.NewReader(content))
	var (
		t  TestCase
		ts []TestCase
	)
	for {
		err := decoder.Decode(&t)
		if errors.Is(err, io.EOF) {
			break
		}
		if err != nil {
			return err, nil
		}
		ts = append(ts, t)
	}
	return nil, ts
}

func cwd() string {
	_, b, _, _ := runtime.Caller(0)
	return filepath.Dir(b)
}
